src/cscapi/sql_storage.py

```diff
@@ -143,49 +143,49 @@ class SQLStorage(storage.StorageInterface):
     def __init__(self, connection_string="sqlite:///cscapi.db") -> None:
         engine = create_engine(connection_string, echo=False)
         Base.metadata.create_all(engine)
-        Session = sessionmaker(bind=engine)
-        self.session = Session()
```

Comment:
If we believe chatGPT: 

The `Session()` call creates a new SQLAlchemy Session, which opens a new transaction. It's not recommended to start a transaction in the __init__ method.

The reason is that the __init__ method is called when an instance of the class is created. If you start a transaction in the __init__ method, the transaction will be open for the entire lifetime of the object, which is not what you typically want. Transactions should be short-lived and should be started just before performing database operations, and ended (committed or rolled back) as soon as possible.  

Instead, you should create the sessionmaker in the __init__ method and then start a new session in each method that performs database operations, using a context manager (with statement). This ensures that each session is properly closed, even if an error occurs during the database operations.